service/controller/resource/deployment/resource.go

```diff
@@ -107,24 +106,26 @@ func getDeploymentByName(list []*v1.Deployment, name string) (*v1.Deployment, er
 	return nil, microerror.Mask(notFoundError)
 }
 
-func isDeploymentModified(a, b *v1.Deployment) bool {
-	aVersion, ok := a.GetAnnotations()[key.VersionBundleVersionAnnotation]
-	if !ok {
-		return true
-	}
+func isAnnotationModified(a, b *v1.Deployment, annotation string) bool {
+	aVersion := a.GetAnnotations()[annotation]
 	if aVersion == "" {
 		return true
 	}
 
-	bVersion, ok := b.GetAnnotations()[key.VersionBundleVersionAnnotation]
-	if !ok {
+	bVersion := b.GetAnnotations()[annotation]
+	if bVersion == "" {
 		return true
 	}
-	if bVersion == "" {
+
+	return aVersion != bVersion
+}
+
+func isDeploymentModified(a, b *v1.Deployment) bool {
+	if isAnnotationModified(a, b, key.VersionBundleVersionAnnotation) {
```

Comment:
It seems to me like this will cause us to delete and recreate a deployment when the release annotation is changed (even if core components didn't change). Is this correct? See https://github.com/giantswarm/kvm-operator/blob/master/service/controller/resource/deployment/update.go#L134

Reply:
Sorry to get stuck on this, just making sure that we don't roll nodes when it's not expected. Looking at https://github.com/giantswarm/kvm-operator/blob/131ae10ae4e055650fb75840d5f3ea2597d4f25d/service/controller/resource/deployment/update.go#L33 I see that we update the Kubernetes resource when it comes from `patch.SetUpdateChange(update)` so I agree that it should be fine. I assume you have tested it as well. Let's go with this.